File: 3354-make-array-elements-equal-to-zero/3354-make-array-elements-equal-to-zero.java
class Solution {
    public int countValidSelections(int[] nums) {
        int total = 0, ls = 0, cases = 0;
        for (int num : nums) total += num;

        for (int num : nums) {
            int rs = total - ls - num;
            if (num == 0) {
                if (ls == rs) {
                    cases += 2;
                } else if (ls == rs - 1 || ls - 1 == rs) {
                    cases += 1;
                }
            }
            ls += num;
        }
        return cases;
    }
}

File: 3354-make-array-elements-equal-to-zero/3354-make-array-elements-equal-to-zero.py
class Solution(object):
    def countValidSelections(self, nums):
        total, ls, cases = sum(nums), 0, 0
        for i in range(len(nums)):
            rs = total - ls - nums[i]
            if nums[i] == 0:
                if ls == rs:
                    cases += 2
                elif ls == rs - 1 or ls - 1 == rs:
                    cases += 1
            ls += nums[i]
        return cases